src/components/AnalysisPage/UnitsSelector/UnitsSelector.jsx

```diff
@@ -1,20 +1,43 @@
 import { useState } from "react";
 import { Select, Space } from "antd";
+import { CheckCircleOutlined } from "@ant-design/icons";
 import "./unitsSelector.css";
 import PropTypes from "prop-types";
 
 const UnitsSelector = ({ partId }) => {
   const [units, setUnits] = useState(undefined);
-  console.log("partId: ", partId);
+  const [showSuccess, setShowSuccess] = useState(false);
 
-  const handleChange = (value) => {
+  const handleChange = async (value) => {
     setUnits(value);
-    console.log(`selected ${value}`);
+    try {
+      const response = await fetch(`/api/v0/parts/${partId}/units`, {
+        method: "PUT",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({ units: value }),
+      });
+
+      if (!response.ok) {
+        throw new Error("Failed to update units.");
+      }
+
+      const data = await response.json();
+      console.log(data);
+
+      setShowSuccess(true);
+      setTimeout(() => {
+        setShowSuccess(false);
+      }, 2000);
+    } catch (error) {
+      console.error("Error updating units:", error);
```

Comment:
It might be good to pop up a little message saying that it failed to update units

Reply:
@rknizzle 
I added a little emoji  in the popup that we do not have to keep! I just wasn't sure what the next step should be for the user after being notified that the units didn't update. 

Should they be able to report an issue somehow?

Reply:
Yea good question. I just put in a little message saying to try again or report it to my email.
Here are my commits if you're curious: 
[11c9afc](https://github.com/rknizzle/propermesh-view/pull/37/commits/11c9afcbe49423ee8bfc27df83aa385234038da8)
[c1025a3](https://github.com/rknizzle/propermesh-view/pull/37/commits/c1025a3415096f0fa42b2a67d4806a9c3abd7acf)

Reply:
Also I think just the floating error message is good enough rather than needing to press ok to get out of the modal